### PR TITLE
added Ironclaw GUI mouse keys & LEDs support

### PR DESCRIPTION
**Pull request by davidhiendl**
first step for Ironclaw mouse GUI #345

### CODE REVIEW

File: src/gui/keymap.cpp [hunk 1]
```diff
@@ -847,6 +877,19 @@ static QHash<QString, Key> getMap(KeyMap::Model model, KeyMap::Layout layout){
         }
         break;
     }
+    case KeyMap::IRONCLAW:{
+        // M65 isn't a keyboard; all mouse maps are unique.
```
Comment (tatokis) [inline src/gui/keymap.cpp]: Please fix this comment

File: src/gui/keymap.cpp [hunk 2]
```diff
@@ -1273,6 +1322,9 @@ QString KeyMap::friendlyName(const QString& key, Layout layout){
     if(map.contains(key))
         return map[key].friendlyName();
     map = KeyMap(HARPOON, layout);
+    if(map.contains(key))
```
Comment (tatokis) [inline src/gui/keymap.cpp]: These changes are not needed

File: src/gui/keymap.cpp [hunk 3]
```diff
@@ -393,6 +393,36 @@ static const Key ST100Zones[] = {
...
+    {0, "Logo",          "back",       22, 30,  8, 28, true,  false },
+    {0, "Wheel",         "wheel",      22, 10,  8,  8, true,  false },
+    
+    // need to add DPI LED, even if not directly configurable for indicator to work
+    {0, "DPI",           "dpi",        10, 10,  5,  5, true,  false }
+
+};
+#define KEYCOUNT_IRONCLAW    (sizeof(IronclawKeys) / sizeof(Key))
+
+#define IRONCLAW_WIDTH       52
```
Comment (tatokis) [inline src/gui/keymap.cpp]: These are unused

**Comment by tatokis** [comment]
You might also want to add the email address you used to make this commit to github, so that it recognises it as yours.

File: src/gui/keymap.cpp (hunk 2 above)
Comment (davidhiendl) [reply src/gui/keymap.cpp]: Are you sure? As I understand it this is the lookup for getting human-friendly name for mouse-specific keys?

File: src/gui/keymap.cpp (hunk 3 above)
Comment (davidhiendl) [reply src/gui/keymap.cpp]: const KEYCOUNT is used for for-loop, WIDTH/HEIGHT are placeholders for when the image is added.

**Comment by MordicusEtCubitus** [comment]
Hi,

I have open this ticket (not able to add :new-device) to get support for the IRONCLAW mice:
https://github.com/ckb-next/ckb-next/issues/558

Is it working yet ? Actually ckb-next (0.4.2 just compiled) does not detect it on my linux (kernel 5.3)
See my comment and device description.

Do not care about wireless but managing colors and dpi

**Comment by MordicusEtCubitus** [comment]
Regarding defect #558 I can provide more details:

* First I had an old version installed by apt on my linux mint, so I removed it
* Then run the scripts compiles with /usr/local/bin/ckb-next  
   Device was not detected
* In file usb.h, I've replaced P_IRONCLAW with my number 
```
// #define P_IRONCLAW           0x1b5d
#define P_IRONCLAW           0x1b4c
```
Recompiling...
Then device was found, but I cannot manage anything, and mouse cursor was no more moving

Here are messages from daemon:

```
$ sudo ckb-next-daemon 
ckb-next: Corsair RGB driver 0.4.2-51-g50865dc
[I] Root controller ready at /dev/input/ckb0
[I] Connecting CORSAIR IRONCLAW RGB WIRELESS Gaming Mouse at /dev/input/ckb1
[I] Starting input thread for /dev/input/ckb1
[W] os_inputmain (usb_linux.c:371): Unable to read wMaxPacketSize for /ep_81, assuming 64
[W] os_inputmain (usb_linux.c:371): Unable to read wMaxPacketSize for /ep_82, assuming 64
[W] os_usbrecv (via firmware.c:24): ckb1: Timeout while waiting for response
[W] _start_dev (device.c:43): Unable to load firmware version/poll rate
[I] Setup finished for /dev/input/ckb1
[E] process_input_urb (keymap.c:362): Unknown mouse data received in input thread 00 from endpoint 82
[E] process_input_urb (keymap.c:362): Unknown mouse data received in input thread 00 from endpoint 82
```

**Pull request merged**